pagefaultgames/pokerogue
# [QoL] Show currently bound controls on starter menu for last active device

## What are the changes?
The starter screen will now show the appropriate keybinds based on the input device you are using, rather than a static string

## Why am I doing these changes?
I play mostly on mobile with controller, but also on PC with controller on occasion. Prior to these changes, there was no immediate way to know (from a UX perspective) which buttons on the controller corresponded to which toggles in the starter screen. This PR aims to fix that by polling input from the last device and dynamically updating the control icons to reflect it.

## What did change?
There was a significant change to starter-select-ui-handler::updateInstructions() in order to accomodate the new UI changes

### Screenshots/Videos
Prior to my changes: ![image](https://github.com/pagefaultgames/pokerogue/assets/170910983/ff0a0a2a-3c0e-497e-83c0-60849feccaf9)

After my changes (keyboard example): 
![image](https://github.com/pagefaultgames/pokerogue/assets/170910983/f37e9759-70d4-466f-86a6-346c1d390a08)

After my changes (controller example):
![image](https://github.com/pagefaultgames/pokerogue/assets/170910983/4bd44973-c14a-474b-8dc9-7e80b77558cd)

## How to test the changes?
- Select new game
- ensure you have a starter with multiple genders/natures/abilities unlocked
- connect an external controller to your device OR simply use your keyboard
- navigate the menu with one or both options to see the icons dynamically update based on your preferred input
- rebinds are also supported and should update the icons

## Checklist

![image](https://github.com/pagefaultgames/pokerogue/assets/170910983/51c0405d-ca54-4981-a8f3-eed7d70014be)
All unit tests passed
